packages/pn-personafisica-webapp/src/redux/sidemenu/actions.ts

```diff
@@ -1,37 +1,92 @@
+  DigitalAddress,
+  DigitalAddresses,
+  IOAllowedValues,
+} from '../../models/contacts';
+import { setProfilePropertyValues } from '../../utility/mixpanel';
 import { Delegator } from '../delegation/types';
 
-export enum SIDEMENU_ACTIONS  {
+export enum SIDEMENU_ACTIONS {
   GET_SIDEMENU_INFORMATION = 'getSidemenuInformation',
 }
 
+/**
+ * Function to get the profile addresses data for mixpanel profile properties
+ * @param addresses
+ */
+const getProfileAddressesData = (addresses: DigitalAddresses): ProfileAddresses => {
+  const hasPec = addresses.legal.length > 0 ? 'yes' : 'no';
+  const hasEmail = addresses.courtesy.some(
+    (address) => address.channelType === CourtesyChannelType.EMAIL
+  )
+    ? 'yes'
+    : 'no';
+  const hasSMS = addresses.courtesy.some(
+    (address) => address.channelType === CourtesyChannelType.SMS
+  )
+    ? 'yes'
+    : 'no';
+  const hasAppIO = addresses.courtesy.some(
+    (address) => address.channelType === CourtesyChannelType.IOMSG
+  )
+    ? 'activated'
+    : 'deactivated';
+
+  return {
+    hasPec,
+    hasEmail,
+    hasSMS,
+    hasAppIO,
+  };
+};
+
 export const getSidemenuInformation = createAsyncThunk<Array<Delegator>>(
   SIDEMENU_ACTIONS.GET_SIDEMENU_INFORMATION,
   // performThunkAction(() => DelegationsApi.getDelegators())
   async () => {
     try {
-      return await DelegationsApi.getDelegators();
+      const delegators = await DelegationsApi.getDelegators();
+      setProfilePropertyValues('profile', 'SEND_HAS_MANDATE', delegators.length > 0 ? 'yes' : 'no');
+      return delegators;
     } catch (e) {
       return [];
     }
   }
 );
 
 // PN-7095 - per capire quale categorie di recapito far vedere nel DomicileBanner
-// si devono prendere gli indirizzi default, tranne per AppIO che si prendono tutti (senza verificare default) 
+// si devono prendere gli indirizzi default, tranne per AppIO che si prendono tutti (senza verificare default)
 // il cui valore non sia DISABLED (cfr. description e commenti della issue JIRA)
 export const getDomicileInfo = createAsyncThunk<Array<DigitalAddress>>(
   'getDomicileInfo',
   performThunkAction(async () => {
-    const isDefaultAddress = (address: DigitalAddress) => 
-      (address.channelType !== CourtesyChannelType.IOMSG && address.senderId === 'default') 
-      || 
-      (address.channelType === CourtesyChannelType.IOMSG && address.value !== IOAllowedValues.DISABLED);
+    const isDefaultAddress = (address: DigitalAddress) =>
+      (address.channelType !== CourtesyChannelType.IOMSG && address.senderId === 'default') ||
+      (address.channelType === CourtesyChannelType.IOMSG &&
+        address.value !== IOAllowedValues.DISABLED);
     const allAddresses = await ContactsApi.getDigitalAddresses();
-    return [...allAddresses.legal.filter(isDefaultAddress), ...allAddresses.courtesy.filter(isDefaultAddress)];
-  }
-));
+
+    const { hasPec, hasEmail, hasSMS, hasAppIO } = getProfileAddressesData(allAddresses);
+
```

Comment:
we can use the same mechanism used with trackingMiddleware function to intercept the response from async http call. Avoid putting extra logic related to other areas directly in async actions